### PR TITLE
Delete method for Questions

### PR DESCRIPTION
**Pull request by vlasovigor**
- [x] Method

- [x] Test

- [x] ACL

- [x] Test ACL

- [x] Exceptions
https://github.com/malinink/easy-tests/issues/374

### CODE REVIEW

**Comment by vlasovigor** [comment]
@malinink ready for review

File: src/main/java/easytests/api/v1/controllers/QuestionsController.java [hunk 1]
```diff
@@ -37,8 +40,28 @@
...
+            throws NotFoundException {
+        final QuestionModelInterface questionModel = this.questionsService.find(id, questionOptions);
+        if (questionModel == null) {
+            throw new NotFoundException();
+        }
+        return questionModel;
+    }
+
+    private QuestionModelInterface getQuestionModel(Integer id) throws NotFoundException {
+        return this.getQuestionModel(id, this.questionsOptionsBuilder.forDelete());
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/QuestionsController.java]: nope

File: src/main/java/easytests/api/v1/controllers/QuestionsController.java [hunk 2]
```diff
@@ -37,8 +40,28 @@
...
-     */
+    @DeleteMapping("/{questionId}")
+    public void delete(@PathVariable Integer questionId) throws NotFoundException, ForbiddenException {
+        final QuestionModelInterface questionModel = this.getQuestionModel(questionId);
+
+        if (!this.acl.hasAccess(questionModel)) {
+            throw new ForbiddenException();
+        }
+        
+        this.questionsService.delete(questionModel);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/QuestionsController.java]: nope, that command only delete question. without answers, and deletion will fail on FK constraint.

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by vlasovigor** [comment]
@malinink ready for review

File: src/main/java/easytests/api/v1/controllers/QuestionsController.java [hunk 3]
```diff
@@ -37,8 +50,32 @@
...
+        final QuestionModelInterface questionModel = this.getQuestionModel(questionId);
+
+        if (!this.acl.hasAccess(questionModel)) {
+            throw new ForbiddenException();
+        }
+
+        final List<AnswerModelInterface> answerModels = this.answersService.findByQuestion(questionModel,
+                answersOptionsBuilder.forDelete());
+
+        this.answersService.delete(answerModels);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/QuestionsController.java]: remove

File: src/main/java/easytests/api/v1/controllers/QuestionsController.java [hunk 4]
```diff
@@ -37,8 +50,32 @@
...
+        this.questionsService.delete(questionModel);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/QuestionsController.java]: use forDelete option

File: src/test/java/easytests/api/v1/controllers/QuestionsControllerTest.java [hunk 5]
```diff
@@ -17,8 +77,69 @@
     /**
      * show(issueId)
      */
-    /**
-     * delete(issueId)
-     */
+    @Test
+    public void testDeleteSuccess() throws Exception {
+        final QuestionModelInterface questionModel = this.questionsSupport.getModelFixtureMock(0);
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/QuestionsControllerTest.java]: перемудрили, не нужна тут модель

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by vlasovigor** [comment]
@malinink pls,look

**Review by malinink — APPROVED** [review]

**Pull request merged**